src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,817 @@
+) -> None:
+    checkpoint_file = Path(checkpoint_file)
+    state = dict(
+        benchmark=benchmark,
+        network=network.state_dict(),
+        size=list(network.parameters())[0].size()[1],
+        h=h.tolist(),
+    )
+
+    tmp_file = checkpoint_file.with_suffix(".tmp")
+    with open(tmp_file, "wb") as file:
+        pickle.dump(state, file, protocol=pickle.DEFAULT_PROTOCOL)
+    tmp_file.rename(checkpoint_file)
+
+
+# BUG in pybnn/util/layers.py, where they have self.log_var be a
+# DoubleTensor while x is a FloatTensor! Here we overwrite the
+# forward method, instead of editing the code directly.
+from pybnn.util.layers import AppendLayer
+
+
+@functools.wraps(AppendLayer.forward)
+def _patched_forward(self, x):
+    log_var = self.log_var.type_as(x)
+    return torch.cat((x, log_var * torch.ones_like(x)), dim=1)
+
+
+AppendLayer.forward = _patched_forward
+
+
+InputType = TypeVar("InputType")
+
+
+class ProfetTask(BaseTask, Generic[InputType]):
+    def __init__(
+        self,
+        benchmark: str,
+        max_trials: int = 100,
+        task_id: int = 0,
+        seed: int = 123,
+        input_dir: Union[Path, str] = None,
+        checkpoint_dir: Union[Path, str] = None,
+        train_config: MetaModelTrainingConfig = None,
+        device: Union[torch.device, str] = None,
+    ):
+        super().__init__(max_trials=max_trials)
+        self.benchmark = benchmark
+        self.task_id = task_id
+        self.seed = seed
+        self.input_dir = Path(input_dir if input_dir else "profet_data")
+        self.checkpoint_dir = Path(checkpoint_dir or self.input_dir / "checkpoints")
+        # The config for the training of the meta-model.
+        # NOTE: the train config is used to determine the hash of the task.
+        # TODO: Also save the train_config in the `configuration` of the task.
+        self.train_config = train_config or MetaModelTrainingConfig()
+
+        # The parameters that have an influence over the training of the meta-model are used to
+        # create the filename where the model will be saved.
+        task_hash_params = dict(
+            benchmark=self.benchmark,
+            task_id=self.task_id,
+            seed=self.seed,
+            **asdict(self.train_config),
+        )
+        logger.info(f"Task hash params: {task_hash_params}")
+        task_hash = compute_identity(**task_hash_params)
+
+        filename = f"{task_hash}.pkl"
+
+        self.checkpoint_file = self.checkpoint_dir / filename
+        logger.info(f"Checkpoint file for this task: {self.checkpoint_file}")
+
+        self.device: torch.device
+        if isinstance(device, torch.device):
+            self.device = device
+        else:
+            self.device = torch.device(
+                device or ("cuda" if torch.cuda.is_available() else "cpu")
+            )
+
+        # TODO: Save those in the configuration dict?
+        self._np_rng: Optional[np.random.RandomState] = None
```

Comment:
Saving what? The RNG state?

Reply:
yes

Reply:
No, that would mean saving the state of the object, not the configuration. The seed should be enough.

Reply:
Oh well, for algorithms we use re-play so using the seed is fine. In this case we don't. :thinking: In usual tasks (not benchmark), the user would be responsible for saving the state and resuming. If we need to save the state we should save it separately from the configuration to be clean. I think that is a discussion worth having in a separate issue as it involves (potentially) all tasks.

Reply:
Issue: https://github.com/Epistimio/orion/issues/665

Reply:
You can remove the TODO